syntax = "proto3";

package stock.v1;

message StockPrice {
  optional float opening_price = 1;
  optional float high_price = 2;
  optional float low_price = 3;
  optional float close_price = 4;
  optional float adjust_close_price = 5;
  string datetime = 6;
}

message GetStockPricesRequest {
  string ticker = 1;
  string start_date = 2;
  string end_date = 3;
}

message GetStockPricesResponse {
  repeated StockPrice stock_prices = 1;
}

service StockService {
  rpc GetStockPrices(GetStockPricesRequest) returns (GetStockPricesResponse) {}
}
